web/trace.go

```diff
@@ -0,0 +1,40 @@
+package web
+
+import (
+	"log"
+
+	"github.com/google/zoekt"
+	"github.com/google/zoekt/query"
+	"github.com/google/zoekt/trace/ot"
+	"github.com/opentracing/opentracing-go"
+	"golang.org/x/net/context"
+)
+
+// traceAwareSearcher wraps a zoekt.Searcher instance so that the tracing context item is set in the
+// context. This context item toggles on trace collection via the
+// github.com/sourcegraph/zoekt/trace/ot package.
+type traceAwareSearcher struct {
```

Comment:
This struct wraps the underlying `zoekt.Searcher` instance. The only handler that does anything is `Search` (the other methods are just pass-throughs). `Search` adds 2 items to the context: (1) whether we should trace at all, according to `frontend` and (2) the SpanContext if we are tracing.